contracts/kleros/KlerosLiquid.sol

```diff
@@ -0,0 +1,724 @@
+            require(dispute.appealDraws[dispute.appealDraws.length - 1] == dispute.votes[dispute.votes.length - 1].length, "The dispute has not finished drawing yet.");
+            dispute.period = courts[dispute.subcourtID].hidden ? Period.commit : Period.vote;
+        } else if (dispute.period == Period.commit) {
+            require(
+                // solium-disable-next-line security/no-block-members
+                block.timestamp - dispute.lastPeriodChange >= courts[dispute.subcourtID].timesPerPeriod[uint(dispute.period)] || dispute.appealCommits[dispute.appealCommits.length - 1] == dispute.votes[dispute.votes.length - 1].length,
+                "The commit period time has not passed yet and not every juror has committed yet."
+            );
+            dispute.period = Period.vote;
+        } else if (dispute.period == Period.vote) {
+            require(
+                // solium-disable-next-line security/no-block-members
+                block.timestamp - dispute.lastPeriodChange >= courts[dispute.subcourtID].timesPerPeriod[uint(dispute.period)] || dispute.appealVotes[dispute.appealVotes.length - 1] == dispute.votes[dispute.votes.length - 1].length,
+                "The vote period time has not passed yet and not every juror has voted yet."
+            );
+            dispute.period = Period.appeal;
+        } else if (dispute.period == Period.appeal) {
+            // solium-disable-next-line security/no-block-members
+            require(block.timestamp - dispute.lastPeriodChange >= courts[dispute.subcourtID].timesPerPeriod[uint(dispute.period)], "The appeal period time has not passed yet.");
+            dispute.period = Period.execution;
+        } else if (dispute.period == Period.execution) {
+            revert("The dispute is already in the last period.");
+        }
+
+        // solium-disable-next-line security/no-block-members
+        dispute.lastPeriodChange = block.timestamp;
+        emit NewPeriod(_disputeID, dispute.period);
+    }
+
+    /** @dev Sets the caller's stake in a subcourt.
+     *  @param _subcourtID The ID of the subcourt.
+     *  @param _stake The new stake.
+     */
+    function setStake(uint _subcourtID, uint _stake) external onlyDuringPhase(Phase.staking) {
+        require(
+            _stake == 0 || courts[_subcourtID].minStake <= _stake,
+            "The juror's stake cannot be lower than the minimum stake for the subcourt."
+        );
+        uint _stakeDiff = _stake - stakeOf(bytes32(_subcourtID), msg.sender);
+        require(
+            _stake == 0 || pinakion.balanceOf(msg.sender) >= stakeOf(bytes32(0), msg.sender) + _stakeDiff,
+            "The juror's total amount of staked tokens cannot be higher than the juror's balance."
+        );
+
+        Juror storage juror = jurors[msg.sender];
+        if (_stake == 0) {
+            if (juror.currentSubcourtIDSMap[_subcourtID]) juror.currentSubcourtIDSMap[_subcourtID] = false;
```

Comment:
We need to remove from `subcourtIDS` as well right?

Reply:
It would be too gas intensive. Better to leave it and then when looping in the front end you check `currentSubcourtIDSMap` to see if you should consider it or not.